File: Baekjoon/2526.cpp
#include <iostream>
#include <set>
using namespace std;

int main() {
	int n, p;
	cin >> n >> p;
	int num = n;
	set<int> s;
	set<int> overlap;
	while (true) {
		num *= n;
		num %= p;
		if (s.find(num) != s.end()) {
			if (overlap.find(num) != overlap.end()) break;
			else overlap.insert(num);
		}
		else {
			s.insert(num);
		}
	}
	cout << overlap.size() << '\n';
	return 0;
}

File: Baekjoon/2526.py
import sys

n, p = map(int, sys.stdin.readline().rstrip().split())

s = set()
overlap = set()
num = n
while True:
    num *= n
    num %= p
    if num in s:
        if num in overlap:
            break
        else:
            overlap.add(num)
    else:
        s.add(num)

print(len(overlap))
